daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -107,8 +107,20 @@ object SExpr {
   final case class SEMakeClo(fv: Array[Int], arity: Int, body: SExpr)
       extends SExpr
       with SomeArrayEquals {
+
     def execute(machine: Machine): Ctrl = {
-      Ctrl.fromPrim(PClosure(body, fv.map(machine.getEnv)), arity)
+      def convertToSValues(fv: Array[Int], getEnv: Int => SValue) = {
```

Comment:
Looks worse, but Yourkit profiler showed that calling `map` on `fv` overly abused `ArrayOps` with turned out to be terribly slow. So this old fashion way of iterating over array has improved speed dramatically.

Reply:
As a rule of the thumb, I would like to have real benchmarks for this kinds of changes. 
What do you think @dajmaki ?

Reply:
I'm fine with these type of changes. Would definitely be interested in hearing how big of an improvement this was.

Reply:
if @dajmaki  is happy, I am happy.

Reply:
Some results from ScalaMeter (this performance pit was actually found using YourKit, I'm using ScalaMeter just to prove a point):
1. Iterating through array old-fashion way
```
      val length = arr.length
      var i = 0
      while (i < length) {
        arr(i)
        i += 1
      }
```
took `Parameters(size -> 15000000): 3.17E-4 ms`

2. Iterating through array using `map`
      ```map.map(_ => ())```
took `Parameters(size -> 15000000): 67.933929 ms`

The difference is huge, which is even more crucial for this code as it runs so many times.